src/net_processing.cpp

```diff
@@ -2384,10 +2374,10 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
                 Misbehaving(pfrom->GetId(), nDoS);
             }
         }
+        return true;
     }
 
-
-    else if (strCommand == NetMsgType::CMPCTBLOCK && !fImporting && !fReindex) // Ignore blocks received while importing
+    if (strCommand == NetMsgType::CMPCTBLOCK && !fImporting && !fReindex) // Ignore blocks received while importing
```

Comment:
May be out of scope for this PR, but I think it'd be better for these conditionals to be nested:

```c++
if (strCommand == NetMsgType::CMPCTBLOCK) {
    if (!fImporting && !fReindex) { // Ignore blocks received while importing
        ...
    }
    // could optionally log in an else branch here
    return true;
}
```

for a couple of reasons:

- it prevents compact blocks from being logged as "Unknown command" if received during import/reindex
- it maintains your new control flow of each message type returning after its own code block.

Same for BLOCKTXN, HEADERS and BLOCK message code blocks below.

Reply:
Indeed out of scope, as this is not refactoring, but bugfixing-like territory.

Reply:
ok